internal/artifact/client.go

```diff
@@ -0,0 +1,53 @@
+package artifact
+
+import (
+	"context"
+	"io"
+	"net/http"
+
+	"github.com/tencentyun/cos-go-sdk-v5"
+)
+
+// Client interface is unstable and may change in the future.
+type Client interface {
+	GetBucket(ctx context.Context, prefix string) (*cos.BucketGetResult, error)
+	PutObject(ctx context.Context, name string, content io.Reader, mimeType string) error
+	GetObject(ctx context.Context, name string) (body io.ReadCloser, header http.Header, err error)
+	DeleteObject(ctx context.Context, name string) error
+}
+
+type cosClient struct {
+	*cos.Client
+}
+
+func NewCosClient(client *cos.Client) Client {
```

Comment:
exported: exported function NewCosClient should have comment or be unexported (revive)